db/migrate/20230215085312_migrate_message_body_to_action_text.rb

```diff
@@ -0,0 +1,10 @@
+class MigrateMessageBodyToActionText < ActiveRecord::Migration[7.0]
+  include ActionView::Helpers::TextHelper
+  def change
+    rename_column :messages, :body, :body_old
+    Message.all.each do |message|
```

Comment:
if you use a `ActiveRecord` in a migration, create it also in the `Migration`, like in the [other files](https://github.com/foodcoops/foodsoft/blob/817e409a2b142acea41520a88d8e7ad7017668a3/db/migrate/20181204070000_create_stock_events.rb#L2-L4). otherwise migration break if the corresponding `ActiveRecord` gets removed from the code in future or the table definition change in an incompatible way

Reply:
I tried to implement it like this and came across a challenge: 
The new message body is a RichText field which writes to the `action_text_rich_texts` table and has a record_type field (Single Table Inheritance). Using a extra class in the migration yields to a record_type like `MigrateMessageBodyToActionText::Message`. In the `up` direction of the migration I could simply update this field to be of type `Message` and that should be okay. But I don't know how to read the body again in the `down` direction, as it would try to find only records with the MigrateMessageBody Type. Any ideas? Is there a way to access the model with a modified `record_type` param?

I've got it working with plain sql, but that feels like a very complicated way. Idk what's better to risk that the Migration stops working if the Message Model gets ever removed or to have a complicated migration file.

```
class MigrateMessageBodyToActionText < ActiveRecord::Migration[7.0]
  include ActionView::Helpers::TextHelper
  class Message < ActiveRecord::Base
    has_rich_text :body
  end

  def change
    reversible do |dir|
      rename_column :messages, :body, :body_old
      dir.up do
        Message.all.each do |message|
          message.update(body: simple_format(message.body_old))
          message.body.update(record_type: :Message) # action_text_rich_texts uses STI record_type field and has to be set to the real model
        end
      end
      dir.down do
        execute "ALTER TABLE `messages` ADD `body_old` text"
        Message.all.each do |message|
          message.update(body_old: message.body.to_trix_html) # message.body is empty here because the record_type doesn't match anymore.
        end
        execute "DELETE FROM `action_text_rich_texts` WHERE `action_text_rich_texts`.`record_type` = 'Message'"
      end
      remove_column :messages, :body_old, :text
    end
  end
end
```

Reply:
I've decided to go wit a custom sql statement for the down path